### ISSUE TITLE
Estimate sensitivity on full LHCb dataset

### ISSUE THREAD

**Issue by MannyMoo** [MannyMoo/AGammaD0Tohhpi0#12]
The final aim is to evaluate what sensitivity we get to zcp and deltaz with the statistics available from the full LHCb dataset. I estimate that we should get around 1M signal D0->pipipi0 candidates (previously I'd thought it was 500k).

Firstly, get the latest world average values for x, y, |q/p| and phi from [here](https://hflav-eos.web.cern.ch/hflav-eos/charm/Moriond19/results_mix_cpv.html) - take the results from the "CPV-allowed" column of the final results table.

Then generate 100 datasets of 1M candidates with those parameters (I tested that the new generator is OK doing 1M).
- One set without any experimental effects.
- One set with experimental effects (probably just decay time efficiency and resolution for now since the phase space efficiency and resolution isn't yet working).

Then run the fits on these datasets and find the average uncertainty on Re(zcp), Im(zcp), Re(dz) and Im(dz) for the datasets without experimental effects and compare this to the dataset with experimental effects. Also check if the experimental effects introduce any (additional) bias to the fit parameters by looking at the pulls. Finally, compare the average uncertainties to the uncertainties from the D0->Kspipi paper.

**Comment by NiallMcHugh** [comment]
OK I'll start on that just now. The only issue is that the batch system looks quite busy just now on the medium queue, and if I recall correctly the jobs with 500k events were taking ~30 mins each so placing them on the short queue might be cutting it a little fine. I'll let you know if this looks like it's going to be a problem in terms of getting results on time once I've run one locally to get a rough idea of runtime.

**Comment by NiallMcHugh** [comment]
Seems to be taking ~45 minutes locally (just passed 500k events after running for slightly more than 20 minutes). Last time the jobs ran locally in ~20 minutes, so assuming about 50% longer again I think they're going to have to go into the medium queue. I'll send them just now and we can see how long they take to run. If we're a little too pushed for time to get the results today then I'm happy to come back in once I'm back from holiday and spend some time pulling everything together.

**Comment by NiallMcHugh** [comment]
I've run the fits yesterday and been able to produce the pull plots, although I've not had any luck copying them to my computer yet (I'm using a Windows machine so there's maybe some other steps I'm missing). I've put them at `/home/ppe/n/nmchugh/SummerProject/` with filenames such as `pull_1M_noExpEff_imDz.png` so I believe you should be able to view them there and it should be easy enough to see what's what.

In terms of the final errors, if I produce a histogram of the error values of each parameter from the `TChain` and take the mean, I'm getting the following results:

**w/o Exp. Effects**
ReDz : 0.00666
ImDz : 0.00173
ReZcp : 0.00667
ImZcp : 0.00173

**w Exp. Effects**
ReDz : 0.00392
ImDz : 0.00102
ReZcp : 0.00392
ImZcp : 0.00102

This seems odd to me, but let me know if these should be calculated differently or I've missed something. I've checked a couple of the `config.txt` files and the `addExpEffects` seems to set correctly to 0 in the "noExpEff" case and 1 in the "wExpEff" case.

**Comment by MannyMoo** [comment]
Yeah, that's the opposite of what I'd expect. Looking at the data, they're as expected for noExpEff vs wExpEff. Are you sure the labels aren't swapped here?

**Comment by MannyMoo** [comment]
Huh, actually, running a quick fit on one of the wExpEff datasets, I see the same thing, the uncertainties are smaller ... This could be because we have the same statistics, but more at higher decay times due to the decay time efficiency, and higher decay times are more sensitive to mixing and CP violation effects.

**Comment by MannyMoo** [comment]
So I guess it's not so meaningful to compare uncertainties with and without the decay-time efficiency for the same statistics. Maybe it'd be more meaningful to generate a large sample without the efficiency, get the uncertainties, then apply the efficiency and cut down the statistics, then compare the uncertainties. Or compare using non-smeared decay time to using smeared decay time.

Anyway, for the final results, you can quote the better uncertainties from the wExpEff datasets :-)

We can still compare if the experimental effects introduce a bias. From the pulls, it looks like things haven't gotten worse, though there's still the bias to Re(zcp): 

- noExpEff:
![pull_1M_noExpEff_imDz](https://user-images.githubusercontent.com/1046916/63174258-82b3c880-c039-11e9-82b8-481dd9f49331.png)
![pull_1M_noExpEff_imZcp](https://user-images.githubusercontent.com/1046916/63174260-82b3c880-c039-11e9-8ed5-de4c0365d735.png)
![pull_1M_noExpEff_reDz](https://user-images.githubusercontent.com/1046916/63174261-82b3c880-c039-11e9-8e90-0a1aeb53e9c8.png)
![pull_1M_noExpEff_reZcp](https://user-images.githubusercontent.com/1046916/63174262-82b3c880-c039-11e9-906b-4ee8ac4cf6cb.png)

- wExpEff:
![pull_1M_wExpEff_imDz](https://user-images.githubusercontent.com/1046916/63174263-82b3c880-c039-11e9-9ec4-8c5979f5519b.png)
![pull_1M_wExpEff_imZcp](https://user-images.githubusercontent.com/1046916/63174265-834c5f00-c039-11e9-815c-0c2e826f9ead.png)
![pull_1M_wExpEff_reDz](https://user-images.githubusercontent.com/1046916/63174267-834c5f00-c039-11e9-8abc-f77b11236361.png)
![pull_1M_wExpEff_reZcp](https://user-images.githubusercontent.com/1046916/63174268-834c5f00-c039-11e9-9516-7c07b728c348.png)
